extensions/rich_text_components/components.py

```diff
@@ -0,0 +1,172 @@
+# coding: utf-8
+#
+# Copyright 2018 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Classes for Rich Text Components in Oppia."""
+
+import re
+
+import bs4
+from extensions.objects.models import objects
+
+
+class BaseRteComponent(object):
+    """Base object class.
+
+    This is the superclass for rich text components in Oppia, such as
+    Image and Video.
+    """
+
+    # These values should be overridden in subclasses.
+    customization_args = {}
+
+    arg_schema = {}
+
+    @classmethod
+    def validate(cls, value_dict):
+        """Validates customization args for a rich text component.
+
+        Raises:
+          TypeError: if any customization arg is invalid.
+        """
+        if cls.arg_schema:
```

Comment:
No, just use one thing and have it be comprehensive. Otherwise this becomes conceptually muddy.

Reply:
Done.